### PR TITLE
[utils] `js_to_json`: Implement template strings

### PR DESCRIPTION
**Pull request by Grub4K**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Rel: https://github.com/yt-dlp/yt-dlp/pull/6505#discussion_r1132854567

Note that I have not really tested against edge cases and only in isolation. More tests are probably required. However I currently lack a lot of time and so just wanted to share this. Feel free to use.

Additionally the current implementation of `template_substitute` works only okayish (`undefined`/`void *` => `null` instead of `undefined`). Generally we would have to provide the real string value of each object which seems out of scope for `js_to_json`.

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [x] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by pukkandan** [comment]
> Note that I have not really tested against edge cases and only in isolation. More tests are probably required.

Limited implementation that covers just our required use case is fine for these kinds of functions

**Comment by Grub4K** [comment]
This passes the common cases in tests and resolves the BibelTV case so should be good to go.

File: test/test_utils.py [hunk 1]
```diff
@@ -1190,6 +1190,12 @@ def test_js_to_json_malformed(self):
         self.assertEqual(js_to_json('42a1'), '42"a1"')
         self.assertEqual(js_to_json('42a-1'), '42"a"-1')
 
+    def test_js_to_json_template_literal(self):
+        self.assertEqual(js_to_json('`Hello ${name}`', {'name': '"world"'}), '"Hello world"')
+        self.assertEqual(js_to_json('`${name}${name}`', {'name': '"X"'}), '"XX"')
+        self.assertEqual(js_to_json('`${name}${name}`', {'name': '5'}), '"55"')
+        self.assertEqual(js_to_json('`${name}"${name}"`', {'name': '5'}), '"5\\"5\\""')
```
Comment (pukkandan) [inline test/test_utils.py:1197]: Add this test

```suggestion
        self.assertEqual(js_to_json('`${name}"${name}"`', {'name': '5'}), '"5\\"5\\""')
        self.assertEqual(js_to_json('`${name}`', {}), '${name}')
```

Comment (Grub4K) [reply test/test_utils.py:1197]: That test will fail though? The expected result (based on the other replacement done) would be `'"name"'`?

Comment (pukkandan) [reply test/test_utils.py:1197]: Ah, right. The main intends of the test is that it shouldn't error
